### PR TITLE
CA1305 String formatting

### PR DESCRIPTION
**Pull request by Rolstenhouse**
Current Culture: Information displayed to user
Invariant Culture: Information written to disk.

- Unsure of which to use where. Or if it would be better to supply a new culture for some

### CODE REVIEW

File: src/Sarif/ConsoleLogger.cs [hunk 1]
```diff
@@ -263,12 +264,12 @@ private string FormatNotificationMessage(Notification notification)
...
             if (!string.IsNullOrEmpty(notification.Id))
             {
-                sb.Append($" {notification.Id}: ");
+                sb.AppendFormat(CultureInfo.CurrentCulture, "$" +notification.Id+": ");
             }
 
             if (!string.IsNullOrEmpty(notification.RuleId))
             {
-                sb.Append($"{notification.RuleId}: ");
+                sb.AppendFormat(CultureInfo.CurrentCulture, "$" + notification.RuleId + ": ");
```
Comment (Rolstenhouse) [inline src/Sarif/ConsoleLogger.cs]: Changing the way the string was generated fixed the error

**Comment by lgolding** [comment]
Very nice. You correctly figured out that:

* You use `InvariantCulture` when constructing a string that's intended to be machine-readable, such as the contents of a SARIF file.
* You use `CurrentCulture` when constructing a string that's intended to be human-readable, such as console output. But...
* You use `InvariantCulture` in exception messages because they're intended for the developer.

(By the way, that last point is debatable, since so often exception messages end up being presented to the end user.)

:+1:

File: src/Sarif/ConsoleLogger.cs (hunk 1 above)
Comment (michaelcfanning) [inline src/Sarif/ConsoleLogger.cs]: You need to get rid of the appended "$" here. It used to broadcast an interpolated string. Or you need to reintroduce the interpolated string literal as the argument to the call (if permitted, I assumed you made a change due to a compilation issue??)

https://msdn.microsoft.com/en-us/library/dn961160.aspx

Comment (lgolding) [inline src/Sarif/ConsoleLogger.cs]: Sorry, I missed this during my review. Michael is right, the "`$`" must be removed. To see what is was there for, do a web search for "c# interpolated strings".

In any case, you don't need to use `AppendFormat` at all. Just call


```
sb.Append(notification.RuleId);
```

... which avoids the whole question of which culture to use.

**Comment by Rolstenhouse** [comment]
Scheduled for merge and close

File: src/Sarif/ConsoleLogger.cs [hunk 2]
```diff
@@ -263,12 +264,12 @@ private string FormatNotificationMessage(Notification notification)
 
             if (!string.IsNullOrEmpty(notification.Id))
             {
-                sb.Append($" {notification.Id}: ");
+                sb.AppendFormat(CultureInfo.CurrentCulture, notification.Id + ": ");
```
Comment (lgolding) [inline src/Sarif/ConsoleLogger.cs]: I think I commented on a previous version of this: you don't need to "format" at all here. You can just write

```
sb.Append(notification.Id);
```
... and avoid the whole issue of "culture".

Comment (Rolstenhouse) [inline src/Sarif/ConsoleLogger.cs]: Is the appended semicolon unnecessary?

Comment (lgolding) [inline src/Sarif/ConsoleLogger.cs]: Oh, I'm sorry! I didn't even see it. I take it all back.

**Pull request merged**